scripts/lldbinit.py

```diff
@@ -11,18 +11,114 @@
 # This script is only meant to be used from LLDB.
 import lldb  # type: ignore
 import os
+import re
+from typing import Any
 
 project_root = os.path.dirname(os.path.realpath(__file__))
 
 ci = lldb.debugger.GetCommandInterpreter()
 result = lldb.SBCommandReturnObject()
 
 
-def RunCommand(cmd: str) -> None:
+def RunCommand(cmd: str, print_command: bool = True) -> Any:
     """Runs a command and prints it to the console to show that it ran."""
-    print("(lldb) %s" % cmd)
+    if print_command:
+        print(f"(lldb) {cmd}")
     ci.HandleCommand(cmd, result)
+    return result.GetOutput()
 
 
 RunCommand(f"settings append target.source-map . {project_root}")
 RunCommand(f"settings append target.source-map /proc/self/cwd {project_root}")
+
+# Matches the output of `print Dump(...)` and captures the stuff from inside the
+# std::string while discarding the std::string type.
+dump_re = re.compile('\\(std::string\\) "((:?.|\n)+)"', re.MULTILINE)
+
+
+# A helper to ease calling the Dump() free functions.
+def cmd_dump(debugger: Any, command: Any, result: Any, dict: Any) -> None:
+    def print_usage() -> None:
+        print("dump <context> [<ID>|<TYPE><ID>|<TYPE> <ID>|-- <ID>]")
+        print("")
+        print('TYPE can be "inst", "entity_name", etc.')
+
+    args = command.split(" ")
+    if len(args) < 2:
+        print_usage()
+        return
+
+    context = args[0]
+
+    # The set of "Make" functions in dump.cpp.
+    id_types = [
```

Comment:
Had you considered making this a dict? That might allow something like:

```
  if m := re.fullmatch("(\w+)\s*(\d+)"):
    id_type = m[0]
    if id_type in id_types:
      id = f"SemIR::{id_types[id_type]}({m[1]})"
```

Note versus the use of `at` below, the above uses `:=` to store the match, and regex capture groups.

Reply:
Oh you have some python skills, thanks, let me try that out.

Reply:
Done! PTAL